content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,314 @@
+The two DAGs are said to be consistent for this message if they also contain the same content for all of its ancestors and progeny and same edges between them.
+
+As time progresses, some messages, by virtue of being encoded in the history of the new message, will be more widely shared throughout the network.
+Once all participants are aware of them, the spread messages and their content are in the scope of global knowledge of the network.
+This is captured by the idea of reachability.
+
+Following the logic of FBA, an important part of exchanging messages between nodes is sharing them not amongst any participant, but only with the important ones for each node.
+This is implementation of FBA federated quorum overlapping structure that aims to reach flexible trust between the groups.
+
+TODO: explain strong reachability
+
+It is crucial that each participant in the netwrok ends up with the same ordering of messages in order to reach a consensus on the ledger history.
+To achieve this stage, participants first compute the slot number of the messages contained in their DAG, and then determine which of the messages are core messages.
+The first messages of each slot are called core messages.
+If a message is a core message and is furthermore not a fork, it is said to be a unique core message.
+TODO: explain better
+
+After each participant has constructed its local DAG and determined core messages, protocol proceeds to [Federated Virtual Voting process](https://drive.google.com/file/d/15adsTes_lhi12PGMxlXfxCsoQsjXvPnS/view).
+
+An FBA construction does not rely on an economic mechanisms for securing consensus because it enables individual nodes to independently make quorum slice decisions.
+Market forces cause overlaps of quorum slice decisions on the network and this gives rise to the network-wide rule for consensus.
+FCP achieves fairness by being both leaderless and totally-ordered, making it excessively difficult for an attacker to influence which of two transactions will be ordered first in a transaction set.
+FCP is also asynchronous and achieves very fast finality.
+These properties make FCP a compelling model for internet-level `Turing-complete` consensus.
+
+## Hash Graph as a Step before FCP
+
+HashGraph is a consensus algorithm that offers a different approach to distributed ledger technology.
+Compared to most blockchain networks, it is a privately held network with its only public version being Hedera Hashgraph network.
+Hashgraph is again a distributed ledger technology and it is described as a continuator or successor to blockchain.
+
+In Hedera, as well as in other blockchains, a user can create a transaction, that will eventually be put in a block and then spread out in the network as a part of this block.
+But, instead of chosing one block to be chained to the previous one forming a chain of blocks, Hedera incorporates every block in the ledger withount discarding any of ret of the records.
```

Comment:
```suggestion
But, instead of choosing one block to be chained to the previous one forming a chain of blocks, Hedera incorporates every block in the ledger without discarding any of ret of the records.
```

Reply:
Corrected https://github.com/optakt/optakt.github.io/pull/21/commits/f524c8d94d32f6edd6da60a9e7f74a848c44d4ce#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R344
Thanks!